File: docs-old-1103-2/extensibility/codesnippet/VisualBasic/walkthrough-implementing-code-snippets_15.vb
        Private Function InsertAnyExpansion(ByVal shortcut As String, ByVal title As String, ByVal path As String) As Boolean
            Dim endColumn As Integer, startLine As Integer
            'get the column number from  the IVsTextView, not the ITextView
            m_vsTextView.GetCaretPos(startLine, endColumn)

            Dim addSpan As New TextSpan()
            addSpan.iStartIndex = endColumn
            addSpan.iEndIndex = endColumn
            addSpan.iStartLine = startLine
            addSpan.iEndLine = startLine

            'get the expansion from the shortcut
            If shortcut IsNot Nothing Then
                'reset the TextSpan to the width of the shortcut, because we're going to replace the shortcut with the expansion
                addSpan.iStartIndex = addSpan.iEndIndex - shortcut.Length

                m_exManager.GetExpansionByShortcut(
                    Me,
                    New Guid(SnippetUtilities.LanguageServiceGuidStr),
                    shortcut,
                    m_vsTextView,
                    New TextSpan() {addSpan},
                    0,
                    path,
                    title)
            End If
            If title IsNot Nothing AndAlso path IsNot Nothing Then
                Dim textLines As IVsTextLines = Nothing
                m_vsTextView.GetBuffer(textLines)
                Dim bufferExpansion As IVsExpansion = DirectCast(textLines, IVsExpansion)

                If bufferExpansion IsNot Nothing Then
                    Dim hr As Integer = bufferExpansion.InsertNamedExpansion(
                        title,
                        path,
                        addSpan,
                        Me,
                        New Guid(SnippetUtilities.LanguageServiceGuidStr),
                        0,
                        m_exSession)
                    If VSConstants.S_OK = hr Then
                        Return True
                    End If
                End If
            End If
            Return False
        End Function

File: docs/extensibility/codesnippet/CSharp/walkthrough-implementing-code-snippets_15.cs
        private bool InsertAnyExpansion(string shortcut, string title, string path)
        {
            //first get the location of the caret, and set up a TextSpan
            int endColumn, startLine;
            //get the column number from  the IVsTextView, not the ITextView
            m_vsTextView.GetCaretPos(out startLine, out endColumn);

            TextSpan addSpan = new TextSpan();
            addSpan.iStartIndex = endColumn;
            addSpan.iEndIndex = endColumn;
            addSpan.iStartLine = startLine;
            addSpan.iEndLine = startLine;

            if (shortcut != null) //get the expansion from the shortcut
            {
                //reset the TextSpan to the width of the shortcut, 
                //because we're going to replace the shortcut with the expansion
                addSpan.iStartIndex = addSpan.iEndIndex - shortcut.Length;

                m_exManager.GetExpansionByShortcut(
                    this,
                    new Guid(SnippetUtilities.LanguageServiceGuidStr),
                    shortcut,
                    m_vsTextView,
                    new TextSpan[] { addSpan },
                    0,
                    out path,
                    out title);

            }
            if (title != null && path != null)
            {
                IVsTextLines textLines;
                m_vsTextView.GetBuffer(out textLines);
                IVsExpansion bufferExpansion = (IVsExpansion)textLines;

                if (bufferExpansion != null)
                {
                    int hr = bufferExpansion.InsertNamedExpansion(
                        title,
                        path,
                        addSpan,
                        this,
                        new Guid(SnippetUtilities.LanguageServiceGuidStr),
                        0,
                       out m_exSession);
                    if (VSConstants.S_OK == hr)
                    {
                        return true;
                    }
                }
            }
            return false;
        }